### ISSUE TITLE
Using this library + Cannot resolve symbol <ID> problem

### ISSUE THREAD

**Issue by MFlisar** [yahoo/squidb#82]
I'm using this library in my project and when I rename a single id in my xml resources, this will lead to 100s of erros in the form of

    error: cannon find symbol class <SQUID ENTRY>

But I can't see the `cannot resolve symbol <ID>` in my messages window in android studio.

Is there some solution for that? I like the library, but if I just rename an id and afterwards have to manually search through all files to search for an highlighted line that's not possible...

**Comment by jdkoren** [comment]
Strange, that doesn't happen for us... Are you compiling with the gradle artifacts, or are you building from source? (If the latter, which branch are you on?) Also, which gradle version are you using?

**Comment by MFlisar** [comment]
I'm using the gradle artifacts (v1.1.0). And I'm using the butterknife library (http://jakewharton.github.io/butterknife/) for view injections. And there is where the problem occurs. The `ids` in the injections are marked read in my projects, but the output does not display them... Instead it outputs all `squidb` as missing whereas android studio visually does not display an problem in those classes...

**Comment by sbosley** [comment]
Yeah, I've seen things like this error come up before. It's actually the same as #72, only with the error caused by a problem in a butterknife annotation instead of a squidb one. From what I've gathered, when there's an error that prevents annotation processing from running at all (e.g. a compilation error in one of the annotations, like a missing id would be for butterknife), then gradle/studio show ALL the compilation errors that result, rather than just the relevant ones that would help you fix the real problem. It's incredibly annoying, and I've been looking for a way to better filter error output since we started the project, but haven't had any luck so far.

That being said, you definitely shouldn't have to search all your files manually for errors. If you use Android Studio's "Refactor" option to rename the ids, it should rename them everywhere (xml and source code). Or a global search for "R.id.old_id" would find everything quickly. Can you use either of those solutions in the short term while we continue to investigate how to make gradle show only relevant errors (or figure out something better)?

**Comment by MFlisar** [comment]
Actually, I had to revert my project once... Was no big deal though. Your "usecases" are good and there I don't have a problem. But following is causing a big problem:

Updating view resources and deleting a view in it which will probably also delete the relevant view id and exactly this one may be used somewhere. Of course, I can search for it before I delete the view, but, sometimes I just don't think about it... And ten minutes later I compile and suddenly I just can't find the root problem anymore...

Another thing, I once thought that the root cause is always somewhere hidden between all the error messages, but I can say for sure, it's not the case. The root cause is not part of the error output (and I did not reach the maximum limit of 200 errors, so this is not the cause for that either)...

That said, currently there's only one solution: being very cautious whenever deleting resources...

**Comment by jdkoren** [comment]
Android studio has a "safe delete" feature as well. In general, use the IDE's functions instead of just editing them directly.

**Comment by MFlisar** [comment]
That's not possible in the xml layout text view, I can't select a `LinearLayout` with it's content and delete it. But your suggestion works, you're right, I don't like to use design view, but there this should work. Although, one case I know where it does not work: if the design view does not work, I can't do it like that. For example, writing a view pager with views in it... This can't be shown in the designer view...

**Comment by sbosley** [comment]
Yeah, that makes sense, it's totally a valid issue. One other (hacky) workaround you could try--make a separate library project for all your model specs and then depend on that library in your main project. In theory, that would force the squidb code generation to happen entirely before compilation of the main project starts so one can't interfere with the other. I know that's not at all an ideal solution so we'll definitely continue to look for something better--maybe there's a way to configure the apt plugin to run a particular processor in isolation first, or maybe we'll write our own gradle plugin to do something like that.
